onnxruntime/core/providers/cuda/tensor/cast_op.cc

```diff
@@ -96,6 +120,8 @@ Status Cast<SrcT>::ComputeInternal(OpKernelContext* context) const {
     CASE(TensorProto_DataType_UINT32, uint32_t)
     CASE(TensorProto_DataType_UINT64, uint64_t)
     CASE(TensorProto_DataType_BOOL, bool)
+    CASE(TensorProto_DataType_FLOAT8E4M3FN, Float8E4M3FN)
```

Comment:
why don't we use `CASE_SAT` here?

Reply:
These lines handles the cast from float 8 to float. saturate is not used. The specialization handles the cast from float to float8. saturate is used here.

Reply:
I thought this is the general template for unspecialized src type. isn't float8 to float handled by `COMPUTE_INTERNAL_FL8`?

Reply:
I should have added  a comment. I added an exception as well. On the CUDA provider, saturate = false is not supported except for float, float16.  The other conversions would not be used. So I did not change the default conversion code.